src/AlgebraicGeometry/Surfaces/K3Auto.jl

```diff
@@ -2040,3 +2044,66 @@ function has_zero_entropy(S::ZZLat; rank_unimod=26)
 
   return maximum(push!([sign(i) for i in d],-1)), data, K3Autgrp, chambers, rational_curves
 end
+
+  
+function _get_pos_entropy_element(gensAutL)
+  G = [first(gensAutL)^0]
+  f = first(gensAutL)^0
+  n = nrows(f)
+  n<= 4 || error("not implemented")
+  while true
+    Gnew = []
+    for g in G
+      for h in gensAutL
+        f = g*h
+        push!(Gnew, f)
+        if rank(f^12-1)==2   # positive entropy
+          return f
+        end
+      end
+    end
+    G = Gnew
+  end
+end
+
+function is_virtually_abelian(L::ZZLat, gensAutL)
```

Comment:
A one-line comment indicating where this algorithm comes from would be really nice to have :-)

(Besides the obvious usefulness for code maintenance, I am in particular interested as I have recently been looking at algorithms testing virtual solvability and virtual nilpotency for matrix groups over number fields and would love to have additional references)

Reply:
Added a reference. Note that this is a very special case: We assume that the matrices act as isometries on a hyperbolic lattice (of signature (1,n)) 
and therefore as isometries on a hyperbolic space. 
These isometries are well understood and we use this. 
For instance the group is finite if and only if it has a fixed point in the interior. It is parabolic (and in particular virtually abelian) if and only if it has a (unique) fixed point on the boundary. It is elementary hyperbolic if and only if it has two fixed points on the boundary. 
(The existence of a fixed point follows from Brouwer's fixed point theorem because the compactification of hyperbolic space used is a compact ball.)